Objects/exceptions.c

```diff
@@ -905,6 +906,11 @@ BaseExceptionGroup_new(PyTypeObject *type, PyObject *args, PyObject *kwds)
         return NULL;
     }
 
+    /* Save initial exceptions sequence as a string incase sequence is mutated */
+    if (!PyList_Check(exceptions) && !PyTuple_Check(exceptions)) {
+        exceptions_str = PyObject_Repr(exceptions);
```

Comment:
This can raise, so you need to check that `exceptions_str != NULL` afterwards.

Reply:
Thanks, this is a good point. It's totally fine if `exceptions_str == NULL`, we just need to handle that case better in the repr.

Reply:
It's not fine. Reraise the exception given by `PyObject_Repr`. If the latter raises this means something went wrong in the C API itself:

```c
exceptions_str = PyObject_Repr(exceptions);
if (exceptions_str == NULL) {
    goto error;
}
```